zebrad/src/commands/start/sync.rs

```diff
@@ -370,19 +401,21 @@
         Ok(())
     }
 
-    /// Queue downloads for each block that isn't currently known to our node
+    /// Queue download and verify tasks for each block that isn't currently known to our node
     async fn request_blocks(&mut self, hashes: Vec<block::Hash>) -> Result<(), Report> {
         tracing::debug!(hashes.len = hashes.len(), "requesting blocks");
         for hash in hashes.into_iter() {
-            // TODO: remove this check once the sync service is more reliable
```

Comment:
I don't think the new comment addresses the concern of the old comment, which is that if we are queuing download of blocks we've already verified, something has gone wrong in the sync process.

The problem isn't that the check needs a better explanation, it's that the check failing indicates an invariant violation.

Reply:
I'll try to answer this question on #980